readthedocs/doc_builder/python_environments.py

```diff
@@ -471,7 +471,13 @@ def install_core_requirements(self):
             Feature.DONT_INSTALL_LATEST_PIP,
             # 20.3 uses the new resolver by default.
             positive='pip<20.3',
-            negative='pip',
+
+            # We are pinning pip to 21.3.1 because builds are failing when
+            # using a newer version. This is a temporal workaround to avoid
+            # builds failing at this step, but we should come back to this and
+            # unpin pip for this case.
+            # https://github.com/readthedocs/readthedocs.org/issues/8864#issuecomment-1025499598
+            negative='pip==21.3.1',
```

Comment:
@stsewd mentioned this might be better as an upper bound selector, to support py27:

```suggestion
            negative='pip<=21.3.1',
```